AsyncDisplayKit/ASCellNode.mm

```diff
@@ -170,14 +170,40 @@ - (void)transitionLayoutWithSizeRange:(ASSizeRange)constrainedSize
 
 - (void)didRelayoutFromOldSize:(CGSize)oldSize toNewSize:(CGSize)newSize
 {
-  if (_layoutDelegate != nil) {
+  if (_interactionDelegate != nil) {
     ASPerformBlockOnMainThread(^{
       BOOL sizeChanged = !CGSizeEqualToSize(oldSize, newSize);
-      [_layoutDelegate nodeDidRelayout:self sizeChanged:sizeChanged];
+      [_interactionDelegate nodeDidRelayout:self sizeChanged:sizeChanged];
     });
   }
 }
 
+- (void)setSelected:(BOOL)selected
+{
+  if (_selected != selected) {
+    _selected = selected;
+    [_interactionDelegate nodeSelectedStateDidChange:self];
```

Comment:
I'm not certain if this will cause a problem, but I just realized that this will call the _interactionDelegate even in the case that the selection is coming from UIKit (rather than a programmer calling setSelected: manually on the node).

One possible result: if a selection had been occurring in an animated way, then passing through the call and defaulting to a non-animated selection could result in overriding the animation / forcing the selection to be instant.

We should test this, and may have to add a way for the _ASTableViewCell / _ASCollectionViewCell to set these values without _interactionDelegate being called.  One way would be to temporarily clear the delegate while setting them, or to have a way to ignore the calls generated during setting them, or to introduce entirely different internal / private ways of setting them that don't do this.  All of them are annoying / shitty :(